src/main/java/FrameworksDrivers/OtherAccount.java

```diff
@@ -60,49 +77,161 @@ public OtherAccount(JPanel masterPanel, CardLayout layout) {
+
+        for (int i = 0; i <= 17; i++) {
+            Label label = new Label();
+            label.createLabelCentered(40, 40, 40, 40, null, "snow", Color.blue);
+            label.setSize(100, 50);
+            this.attributesPanel.add(label.getLabel());
+        }
+
+        attrHolder.add(attributesTitle.getLabel());
+        attrHolder.add(this.attributesPanel);
+
+        //courses
+        JPanel coursesHolder = new JPanel();
+        coursesHolder.setBackground(Color.BLUE);
+        Label coursesTitle = new Label();
+        coursesTitle.createLabelCentered(5, 5, 5, 5, null, "Courses   ", Color.white);
+
+        this.coursesPanel = new JPanel();
+        this.coursesPanel.setLayout(new GridLayout(0, 3));
+
+        for (int i = 0; i <= 4; i++) {
+            Label label = new Label();
+            label.createLabelCentered(40, 40, 40, 40, null, "snow", Color.blue);
+            label.setSize(100, 50);
+            this.coursesPanel.add(label.getLabel());
+        }
+
+        coursesHolder.add(coursesTitle.getLabel());
+        coursesHolder.add(this.coursesPanel);
+
+
         JPanel buttonAlign = new JPanel();
-        buttonAlign.setPreferredSize(new Dimension(600, 20));
-        buttonAlign.setLayout(new BoxLayout(buttonAlign, BoxLayout.X_AXIS));
-        backToMain = new Button();
-        backToMain.createButton(buttonAlign, "Back", 20, 20, 10, 10);
+        buttonAlign.setLayout(new FlowLayout(FlowLayout.LEFT));
+        buttonAlign.setPreferredSize(new Dimension(600, 35));
+        Button backToMain = new Button();
+        backToMain.createButton(buttonAlign, "Back", 30, 30, 30, 30);
         backToMain.getButton().addActionListener(this);
-        backToMain.setHorizontalAlignment("left");
+
 
         this.panel.add(buttonAlign);
         this.panel.add(inner);
         inner.add(upper);
         inner.add(lower);
+        this.panel.add(interestsHolder);
+        this.panel.add(attrHolder);
+        this.panel.add(coursesHolder);
 
+        JScrollPane scroller = new JScrollPane(this.panel,
+                JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
+                JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+        this.panel.setLayout(new BoxLayout(this.panel, BoxLayout.Y_AXIS));
 
-        this.masterPanel.add(this.panel, "otherAccount");
-    }
+        this.masterPanel.add(scroller, "otherAccount");
 
+    }
 
+    /**
+     * Updates
+     *
+     * @parem paths References to pages that OtherAccount may switch to
+     */
     public void sendPaths(Object[] paths) {
         //gets paths to other pages
         this.paths = paths;
     }
 
     @Override
     public void actionPerformed(ActionEvent e) {
-        if(e.getSource() == this.backToMain.getButton()){
-            OtherAccountPresenter otherAccountPresenter = new OtherAccountPresenter(this.name.getLabel().getText());
-            this.layout.show(this.masterPanel, "mainpageView");
+        if (prevpage.equals("main")) {
+
```

Comment:
Is this meant to be empty? If it is, it might be better to get rid of it entirely.